### PR TITLE
Created a Reminder screen, Create reminder screen and the respective …

### PR DESCRIPTION
**Pull request by seomasha**
…viewmodels also a Reminder component and a Reminder model

### CODE REVIEW

File: InternshipApp/InternshipApp/InternshipAppApp.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ struct InternshipAppApp: App {
     
     var body: some Scene {
         WindowGroup {
-            CreateTaskView(taskViewModel: TaskViewModel(), profileViewModel: profileViewModel)
+            CreateTaskView(taskViewModel: TaskViewModel(), profileViewModel: profileViewModel, reminderViewModel: ReminderViewModel(), createReminderViewModel: CreateReminderViewModel())
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/InternshipAppApp.swift]: When initializing something or in any situation where you are passing some params, make sure to format the code in this way: 
Class(firstParam,
           secondParam,
            thirdParam)
So each param has its own row for better code readability

File: InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+//
+//  Created by Sead Mašetić on 14. 8. 2024..
+//
+
+import SwiftUI
+
+struct CreateReminderScreenView: View {
+    
+    @ObservedObject var reminderViewModel: ReminderViewModel
+    @ObservedObject var createReminderViewModel: CreateReminderViewModel
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift]: No need to have two separate ViewModels for one logical unit which is "Reminder". 
Currently you have ReminderViewModel and CreateReminderViewModel, both of those can be combined inside one ViewModel and have everything related to Reminder :)

File: InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            Toggle("Enable recurrence", isOn: $createReminderViewModel.isOn)
+                .padding()
+            
+            Picker("Recurrence:", selection: $createReminderViewModel.selectedOption) {
+                ForEach(createReminderViewModel.options, id: \.self) { option in
+                        Text(option)
+                }
+            }.disabled(!createReminderViewModel.disableRecurrence()).padding()
+            
+            if createReminderViewModel.selectedOption == "Custom" {
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift]: Make these recurring options as Swift Enums, create a separate folder for Enums and inside of it there should be RecurrenceEnums file. 
So when you're comparing them inside statements, you're not doing it by comparing a string, i.e == "custom"

File: InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            }
+
+            
+            Button {
+                if createReminderViewModel.isOn {
+                    if createReminderViewModel.selectedOption == "Custom" {
+                        var formattedString = ""
+
+                        if createReminderViewModel.days > 0 {
+                            formattedString += "\(createReminderViewModel.days) day" + (createReminderViewModel.days > 1 ? "s" : "")
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderScreenView.swift]: Create a method that will do this formatting, so it clears up a view from this logic a bit :)

File: InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import Foundation
+
+final class CreateReminderViewModel: ObservableObject {
+    @Published var reminderTitle: String = ""
+    @Published var reminderDescription: String = ""
+    @Published var reminderDate: Date = Date()
+    
+    @Published var isOn: Bool = false
+    @Published var selectedOption = "Daily"
+    @Published var options = ["Daily", "Weekly", "Monthly", "Custom"]
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/Screens/CreateReminderScreenView/CreateReminderViewModel.swift]: Create a model for option, or an Enum

File: InternshipApp/InternshipApp/Screens/ReminderScreenView/ReminderScreenView.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    @ObservedObject var profileViewModel: ProfileScreenViewModel
+    @ObservedObject var reminderViewModel: ReminderViewModel
+    @ObservedObject var createReminderViewModel: CreateReminderViewModel
+    
+    var body: some View {
+        NavigationStack {
+            ScrollView {
+                VStack {
+                    ForEach(reminderViewModel.reminders, id: \.id) { reminder in
+                        ReminderView(reminderTitle: reminder.title, reminderDescription: reminder.description, reminderDate: reminder.date, reminderRecurrence: reminder.recurrence)
```
Comment (Denis-Tojaga) [inline InternshipApp/InternshipApp/Screens/ReminderScreenView/ReminderScreenView.swift]: Param formatting same thing here, so it doesn't take up the whole line

**Review by denistojaga-symphony — APPROVED** [review]

**Pull request merged**